go/arrow/float16/float16.go

```diff
@@ -29,6 +30,11 @@ type Num struct {
 	bits uint16
 }
 
+var (
+	MaxNum = Num{bits: 0b0111101111111111}
+	MinNum = MaxNum.Negate()
+)
```

Comment:
might make sense to add `func Inf() float16.Num` and `func NaN() float16.Num` functions to this package